src/blueapi/worker/task_worker.py

```diff
@@ -111,33 +126,47 @@ def __init__(
         self._stopped = Event()
         self._stopped.set()
         self._broadcast_statuses = broadcast_statuses
+        self._context_register = {}
+        setup_tracing("BlueAPIWorker", OTLP_EXPORT_ENABLED)
 
-    def clear_task(self, task_id: str) -> str:
+    @start_as_current_span(TRACER, "task_id")
+    def clear_task(self, task_id: str, carr: dict[str, Any] | None = None) -> str:
         task = self._tasks.pop(task_id)
         return task.task_id
 
+    @start_as_current_span(TRACER)
     def cancel_active_task(
         self,
         failure: bool = False,
         reason: str | None = None,
+        carr: dict[str, Any] | None = None,
     ) -> str:
         if self._current is None:
             # Persuades mypy that self._current is not None
             # We only allow this method to be called if a Plan is active
             raise TransitionError("Attempted to cancel while no active Task")
         if failure:
             self._ctx.run_engine.abort(reason)
+            add_span_attributes({"Task aborted": reason})
         else:
             self._ctx.run_engine.stop()
+            add_span_attributes({"Task stopped": reason})
```

Comment:
I would say kno because then you're reporting something that hasn't yet happened.  If they sai stopping rather than stopped then yes, but it seems more useful to know that the  requested operation completed that might complete, to me anyway.